cli/azd/cmd/middleware/middleware.go

```diff
@@ -68,20 +66,9 @@ func (r *MiddlewareRunner) RunChildAction(
 	runOptions *Options,
 	action actions.Action,
 ) (*actions.ActionResult, error) {
-	// If we have previously run this action then return the cached result
-	if cachedActionResult, has := r.actionCache[action]; has {
-		return cachedActionResult, nil
-	}
-
 	// If we have not previously run this action then execute it
 	runOptions.isChildAction = true
 	result, err := r.RunAction(ctx, runOptions, action)
-
-	// Cache the result on action success
-	if err == nil {
-		r.actionCache[action] = result
-	}
-
 	return result, err
```

Comment:
Removing the caching will force the `package` command to run multiple times in the `up` flow.

1. `package`
1. `provision`
1. `package` (via up prereq)
1. `deploy`